doc/source/development/test_writing.rst

```diff
@@ -0,0 +1,103 @@
+.. _test_organization:
+
+Test organization
+=================
+Ideally, there should be one, and only one, obvious place for a test to reside.
+Until we reach that ideal, these are some rules of thumb for where a test should
+be located.
+
+1. Does your test depend only on code in ``pd._libs.tslibs``?
+This test likely belongs in one of:
+
+    - tests.tslibs
+
+        .. note::
+
+             No file in ``tests.tslibs`` should import from any pandas modules outside of ``pd._libs.tslibs``
+
+    - tests.scalar
+    - tests.tseries.offsets
+
+2. Does your test depend only on code in pd._libs?
+This test likely belongs in one of:
+
+    - tests.libs
+    - tests.groupby.test_libgroupby
+
+3. Is your test for an arithmetic or comparison method?
+This test likely belongs in one of:
+
+    - tests.arithmetic
+
+        .. note::
+
+            These are intended for tests that can be shared to test the behavior of DataFrame/Series/Index/ExtensionArray using the ``box_with_array`` fixture.
+
+    - tests.frame.test_arithmetic
+    - tests.series.test_arithmetic
+
+4. Is your test for a reduction method (min, max, sum, prod, ...)?
+This test likely belongs in one of:
+
+    - tests.reductions
+
+        .. note::
+            These are intended for tests that can be shared to test the behavior of DataFrame/Series/Index/ExtensionArray.
+
+    - tests.frame.test_reductions
+    - tests.series.test_reductions
+    - tests.test_nanops
+
+5. Is your test for a DataFrame or Series method?
+    A) Is the method a plotting method?
+    This test likely belongs in one of:
+
+        - tests.plotting
+
+    B) Is the method an IO method?
+    This test likely belongs in one of:
+
+        - tests.io
+
+    C) Is the method an indexing method?
+        i) Is the method ``loc``, ``iloc``, ``at``, or ``iat``?
+        This test likely belongs in one of:
```

Comment:
can you be a little more precise here (e.g. do we have criteria for putting in pandas/tests/indexing over the frame/series ones? (can be a TODO of course)